baseplate/context/thrift.py

```diff
@@ -104,6 +104,12 @@ def _call_thrift_method(self, name, *args, **kwargs):
                             prot.trans.set_header("Sampled", sampled)
                         if span.flags:
                             prot.trans.set_header("Flags", str(span.flags))
+                        if hasattr(span.context, "authentication") \
+                                and span.context.authentication.token:
+                            prot.trans.set_header(
+                                "Authentication",
+                                str(span.context.authentication)
```

Comment:
That makes sense.  Originally I was trying to have the token conversion work via the string casting of the object, but could not get it to handle unset tokens cleanly.  I was hoping to have the case of an unset token handled internal to the wrapper rather than here so that the hand rolled HTTP implementations would not be too hairy.